hasher-matcher-actioner/hmalib/lambdas/pdq/pdq_matcher.py

```diff
@@ -65,25 +52,29 @@ def lambda_handler(event, context):
             continue
         hash_str = message["hash"]
         key = message["key"]
-        query = [hash_str]
         current_datetime = datetime.datetime.now()
-        results = hash_index.search(query, THRESHOLD, return_as_ids=True)
-        # Only checking one hash at a time for now
-        result = results[0]
-        if len(result) > 0:
-            message_str = "Matches found for key: {} hash: {}, for IDs: {}".format(
-                key, hash_str, result
-            )
-            logger.info(message_str)
-            for te_id in result:
-                te_hash = hash_index.hash_at(te_id)
-                save_match_to_datastore(
-                    table, key, te_id, hash_str, current_datetime, te_hash
+        results = hash_index.query(hash_str)
+
+        if len(results) > 0:
+            match_ids = []
+            for match in results:
+                metadata = match.metadata
+                logger.info(
+                    f"Match found for key: {key}, hash: {hash_str} -> {metadata}"
```

Comment:
ignoreable: logger by default supports string formatting, but it's less nice than f-strings, and easier to put the wrong number of arguments, with the possible tradeoff that if the log isn't printed, objects aren't converted to strings

```
logger.info("Match found for key: %s, hash %s -> %s", key, hash_str, metadata)
```